### PR TITLE
fix(duckdb): support large timestamps by way of pyarrow.Table

### PR DESCRIPTION
**Pull request by cpcloud**
This PR fixes support for timestamps that are too big for nanosecond
resolution.

To do this, I had to call `fetch_arrow_table` (before calling `to_pandas` on
_that_) because arrow handles coarser timestamps correctly and pandas doesn't.

A mildly annoying side effect of doing that is that `array<T>`s are now `numpy`
arrays in the resulting `DataFrame` which is why I had to add another `convert`
overload to convert `array<T>` elements back to lists to avoid breaking the
existing API.

pyarrow seems to be the odd one out, since the other array-supporting backends
all return their `array<T>` elements as `list`s.

### CODE REVIEW

**Review by jcrist — COMMENTED** [review]
Small nit, otherwise looks good to me.

**Comment by cpcloud** [comment]
Finally I can reproduce these bizarre never-before-failing failures. It seems to be only reproducible when running with something other than `--numprocesses auto` (for example `--numprocesses 2`) at the moment. I have no idea why the number of cores being "low" would have such a magical effect.

**Comment by cpcloud** [comment]
This requires a fix in pyarrow, so I'm going to hold off on it until there's a release of pyarrow that contains the necessary fix.

**Comment by cpcloud** [comment]
Sadly, this seems to break our TPC-H queries. I'll look into it today.

File: ibis/expr/schema.py [hunk 1]
```diff
@@ -389,7 +389,7 @@ def apply_to(self, df: pd.DataFrame) -> pd.DataFrame:
                 # assume not equal
                 not_equal = True
 
-            if not_equal or isinstance(dtype, (dt.String, dt.Struct)):
+            if not_equal or not isinstance(dtype, dt.Primitive):
```
Comment (kszucs) [inline ibis/expr/schema.py:392]: Seems much cleaner

**Review by kszucs — APPROVED** [review]

**Comment by kszucs** [comment]
@cpcloud can you squash it?

**Comment by cpcloud** [comment]
Squashing!

File: ibis/expr/schema.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/schema.py:392]: Thanks!

**Pull request merged**